kcards/static/room.js

```diff
@@ -1,15 +1,23 @@
-setInterval(function() {
+function checkForUpdates() {
+
+    console.log("Waiting for room updates...")
     $.ajax({
 
-      type: "GET",
-      url: "/api/rooms/" + code + "/timestamp",
-      datatype: "JSON",
+        type: "GET",
+        url: "/api/rooms/" + code + "/timestamp?current=" + timestamp,
+        datatype: "JSON",
 
-      success: function(data) {
-          if (data["timestamp"] != timestamp) {
-            location.reload();
-          };
-      }
+        complete: function(data) {
```

Comment:
`complete` is processed after `success` or `error`: http://api.jquery.com/jquery.ajax/